src/Generator/Passes/PassBuilder.cs

```diff
@@ -32,6 +32,14 @@ public void AddPass(T pass)
             Passes.Add(pass);
         }
 
+        /// <summary>
+        /// Remove a previously-added pass, if exists.
+        /// </summary>
+        public bool RemovePass(T pass)
+        {
+            return pass is TranslationUnitPass && Passes.Remove(pass);
```

Comment:
Any specific reason you are checking for `pass is TranslationUnitPass` here?

Since this method is supposed to be generic, I think keeping the code generic would be best here.